src/MongoDB.Driver/Encryption/LibMongoCryptControllerBase.cs

```diff
@@ -221,6 +232,11 @@ private void SendKmsRequest(KmsRequest request, CancellationToken cancellation)
         {
             var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
             ParseKmsEndPoint(request.Endpoint, out var host, out var port);
+
+            // the way we use this method doesn't work on Linux with .netstandard1.5,
+            // to make it work on that platform, we need to resolve the host and use the IP address.
+            // Since that platform is not supported on libmongocrypt level anyway, instead of fixing it,
+            // we rely on throwing a PlatformNotSupported exception in the previous steps
             socket.Connect(host, port);
```

Comment:
Rather than throwing in each wrapper method via `ThrowIfUnsupportedPlatform()`, why not call `ThrowIfUnsupportedPlatform()` before `socket.Connect(host, port)` and remove the other `ThrowIfUnsupportedPlatform()` calls?

Reply:
1. If we know that the whole operation won't work, why do we need to call it at all?
2. A similar issue happens on mac in a different line (`sslStream.AuthenticateAsClientAsync`)
3. This line is called not in each kms provider (not in the same methods), but anyway it will fail when we call crypto hooks (this particular line is called before hooks), so instead of having different errors in different places, we can standardize this behavior and throw it right at the beginning

Reply:
Fair enough points.